### PR TITLE
Add workflow_dispatch trigger for main build-push-images workflow

### PR DESCRIPTION
**Pull request by tdeebswihart**
I added a workflow_dispatch trigger for our main release action

### CODE REVIEW

**Review by bergundy — APPROVED** [review]

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -25,6 +30,7 @@ jobs:
         uses: actions/checkout@v4
         with:
           submodules: "true"
+          ref: ${{ github.event_name == 'workflow_dispatch' && github.event.inputs.commit || '' }}
```
Comment (yycptt) [inline .github/workflows/docker.yml:33]: nit: I think `github.event.inputs.commit` is empty if the action is not manually started. so the condition might not be necessary.

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -39,15 +45,6 @@ jobs:
...
-        uses: actions/cache/restore@v3
-        with:
-          path: /tmp/.buildx-cache
-          key: ${{ runner.os }}-cache-go-build-${{ hashFiles('**/go.sum') }}
-          restore-keys: |
-            ${{ runner.os }}-cache-go-build-
-
       - name: Prepare build args
         id: build_args
         run: |
```
Comment (yycptt) [inline .github/workflows/docker.yml:50]: This comment is for L54. 
What's the value of `GITHUB_SHA` when the action is manually triggered. From doc it seems it's always the latest commit on a branch which can be different from the input commit hash.

Comment (tdeebswihart) [reply .github/workflows/docker.yml:50]: Ah good point. Perhaps a better option may be to simply skip saving the cache when we manually kick this off; I'll put a separate PR up for that later today

Comment (tdeebswihart) [reply .github/workflows/docker.yml:50]: Fixed in https://github.com/temporalio/docker-builds/pull/171

**Pull request merged**